kuma/users/stripe_utils.py

```diff
@@ -1,141 +1,48 @@
-from datetime import datetime
-
 import stripe
 from django.conf import settings
 from django.utils import timezone
 
 from kuma.core.urlresolvers import reverse
 from kuma.wiki.templatetags.jinja_helpers import absolutify
 
-from . import signals
 from .models import UserSubscription
 
 
-def retrieve_stripe_subscription(customer):
-    """
-    Returns the first subscription it finds matching the configured stripe
-    plan or, if there's none, just the first it finds.
-    """
-    first_subscription = None
-
-    for subscription in customer.subscriptions.list().auto_paging_iter():
-        if first_subscription is None:
-            first_subscription = subscription
-
-        # We have to use array indexing syntax, as stripe uses dicts to
-        # represent its objects (dicts come with an .items method)
-        for item in subscription["items"].auto_paging_iter():
-            if item.plan.id == settings.STRIPE_PLAN_ID:
-                # If we find a subscription matching the selected plan we
-                # return that instead of whatever we found first
-                return subscription
-
-    return first_subscription
-
-
-def retrieve_and_synchronize_subscription_info(user):
-    """For the given user, if it has as 'stripe_customer_id' retrieve the info
-    about the subscription if it's there. All packaged in a way that is
-    practical for the stripe_subscription.html template.
-
-    Also, whilst doing this check, we also verify that the UserSubscription record
-    for this user is right. Doing that check is a second-layer check in case
-    our webhooks have failed us.
-    """
-    subscription_info = None
-    stripe_customer = get_stripe_customer(user)
-    if stripe_customer:
-        stripe_subscription_info = get_stripe_subscription_info(stripe_customer)
-        if stripe_subscription_info:
-            source = stripe_customer.default_source
-            if source.object == "card":
-                card = source
-            elif source.object == "source":
-                card = source.card
-            else:
-                raise ValueError(
-                    f"unexpected stripe customer default_source of type {source.object!r}"
-                )
-
-            subscription_info = {
-                "id": stripe_subscription_info.id,
-                "amount": stripe_subscription_info.plan.amount,
-                "brand": card.brand,
-                "expires_at": f"{card.exp_month}/{card.exp_year}",
-                "last4": card.last4,
-                # Cards that are part of a "source" don't have a zip
-                "zip": card.get("address_zip", None),
-                "next_payment_at": datetime.fromtimestamp(
-                    stripe_subscription_info.current_period_end
-                ),
-            }
+def retrieve_and_synchronize_stripe_subscription(user):
+    if not user.stripe_customer_id:
+        return None
```

Comment:
I plead ignorance here but I suspect the only time this can happen is when we debug. I.e. you mess with the database and switch between test accounts etc. 
If some code calls this function on a user that doesn't have a `stripe_customer_id` it sounds like something is broken in the caller. Perhaps we can inject, before the `return None`, something like `warnings.warn(f"Tried to sync stripe subscription on a user ({user!r}) that doesn't have a stripe_customer_id")`. Or do you think we should throw an error just to feel the most explicit and safe?

Reply:
atm the fn was written with the assumption that it could be called from places where a user has not started setting up a subscription yet and hence also has no customer attached. Explicitly that call site is when fetching the subscription for the /settings endpoint

Reply:
If you have time and are in there in those files, I would suggest you turn that `if` into `assert user.stripe_customer_id` and then conditionally call it like `user.stripe_customer_id and retrieve_and_synchronize_stripe_subscription(user)` at the call site.